test/groupBy.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+
+
+test('groupBy', function() {
+
+  eq(typeof S.groupBy, 'function');
+  eq(S.groupBy.length, 2);
+  eq(S.groupBy.toString(), 'groupBy :: (a -> a -> Boolean) -> Array a -> Array (Array a)');
+
+  var productsOf3 = S.curry2(function(a, b) {
+    return a * b % 3 === 0;
+  });
+
+  eq(S.groupBy(productsOf3, []), []);
+  eq(S.groupBy(productsOf3, [1, 2, 3, 4, 5, 6, 7, 8, 9]), [[1], [2, 3], [4], [5, 6], [7], [8, 9]]);
+  eq(S.groupBy(S.equals, [1, 1, 2, 3, 3, 4]), [[1, 1], [2], [3, 3], [4]]);
+  eq(S.groupBy(S.equals, [1, 1, 2, 1, 1]), [[1, 1], [2], [1, 1]]);
+
+});
```

Comment:
@davidchambers this is working:

```
  jsc.assert(jsc.forall('nat -> nat -> bool', 'array nat', function(f, xs) {
    return S.equals(S.join(S.groupBy(f, xs)), xs);
  }), {tests: 1000});
```

Do you think that's ok? Or would you do something different?